plugin/pkg/admission/podtolerationrestriction/admission_test.go

```diff
@@ -230,57 +215,74 @@ func TestPodAdmission(t *testing.T) {
 		},
 	}
 	for _, test := range tests {
-		if test.namespaceTolerations != nil {
-			tolerationStr, err := json.Marshal(test.namespaceTolerations)
-			if err != nil {
-				t.Errorf("error in marshalling namespace tolerations %v", test.namespaceTolerations)
+		t.Run(test.testName, func(t *testing.T) {
+			namespace := &api.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "testNamespace",
+					Namespace:   "",
+					Annotations: map[string]string{},
+				},
 			}
-			namespace.Annotations = map[string]string{NSDefaultTolerations: string(tolerationStr)}
-		}
 
-		if test.whitelist != nil {
-			tolerationStr, err := json.Marshal(test.whitelist)
-			if err != nil {
-				t.Errorf("error in marshalling namespace whitelist %v", test.whitelist)
+			if test.namespaceTolerations != nil {
+				tolerationStr, err := json.Marshal(test.namespaceTolerations)
+				if err != nil {
+					t.Errorf("error in marshalling namespace tolerations %v", test.namespaceTolerations)
+				}
+				namespace.Annotations = map[string]string{NSDefaultTolerations: string(tolerationStr)}
 			}
-			namespace.Annotations[NSWLTolerations] = string(tolerationStr)
-		}
 
-		informerFactory.Core().InternalVersion().Namespaces().Informer().GetStore().Update(namespace)
```

Comment:
if we were modifying the store directly, how was our list failing to return the updated namespace?

Reply:
Instead of  declaring mockClient independently in every loop, I think the following would be better.
> mockClient := fake.NewSimpleClientset(namespace)

And this line could be:

> mockClient.Core().Namespaces().Update(namespace)

Reply:
> mockClient.Core().Namespaces().Update(namespace)

The informer that is used by the admission controller might not pick up that change, and the test could be seeing the namespace from the previous test case.

Reply:
@liggitt @ericchiang Note that modifying store aka `indexers` will not trigger `Watch` related functionalities of fake client.